### PR TITLE
Server gRPC implementation and abstract class

### PR DESCRIPTION
**Pull request by abonilla-eb**
- Allows gRPC server creation and execution
- Encapsulates gRPC data structures
- Exposes a library agnostic abstract class that is called on gRPC events

### CODE REVIEW

File: edagames_grpc/server.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import grpc
+import eda_games_pb2
+import eda_games_pb2_grpc
+from .game_state import GameState
```
Comment (msotelo-eb) [inline edagames_grpc/server.py]: Avoid relative imports. Use `from edagames_grpc.game_state`

File: edagames_grpc/server.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+from typing import Dict, List
```
Comment (msotelo-eb) [inline edagames_grpc/server.py]: Try to separate local imports from external or library imports.

File: edagames_grpc/server.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    response = eda_games_pb2.GameStateResponse()
+    response.current_player = game_state.current_player
+    response.next_player = game_state.next_player
+    response.turn_data.update(game_state.turn_data)
+    response.play_data.update(game_state.play_data)
+    return response
+
+
+class EdaGamesService(eda_games_pb2_grpc.EdaGameServiceServicer):
+    def __init__(self, delegate):
```
Comment (msotelo-eb) [inline edagames_grpc/server.py]: NIT: Add the delegate's type if possible

Comment (abonilla-eb) [reply edagames_grpc/server.py]: Added in 41a16fc

File: edagames_grpc/server.py (hunk 2 above)
Comment (abonilla-eb) [reply edagames_grpc/server.py]: Changed in e9c9fb8

File: edagames_grpc/server.py (hunk 1 above)
Comment (abonilla-eb) [reply edagames_grpc/server.py]: Changed in e9c9fb8

**Review by ecrespillo-eb — APPROVED** [review]

**Pull request merged**